### PR TITLE
server: telemetry context header

### PR DESCRIPTION
**Pull request by znewton**
## Description

Global TelemetryContext was implemented several major server versions ago. At the same time, the old `getCorrelationId` and `bindCorrelationId` method of tracking correlationId was deprecated. This PR removes usage of those methods, and also adds a new Telemetry Context header that can be extended to track other information for the lifetime of an API request.

For the new `x-telemetry-context` header, the old `x-correlation-id` header will still be respected (for now) if `x-telemetry-context` header does not container `correlationId` property. BasicRestWrapper now takes in an optional `getTelemetryContextProperties` method, similar to how it takes a `getCorrelationId` method. This is used to generate telemetryContext header on outgoing requests from within R11s. `x-correlation-id` is still generated.

## Breaking Changes

- `enableGlobalTelemetryContext` config switched to `true` in code. Was already true in configs.
- `bindCorrelationId` usage was removed from Gitrest, Historian, and Routerlicious Rest APIs, meaning `getCorrelationId` without `enableGlobalTelemetryContext: true` will not work anymore.

I'm leaving the old `getCorrelationId` and `bindCorrelationId` methods in for 1 more release cycle out of abundance of caution, even though it has been deprecated for almost a year.

### CODE REVIEW

**Review by Noelin — APPROVED** [review]

**Review by dhr-verma — APPROVED** [review]

**Comment by tylerbutler** [comment]
@znewton thanks so much for adding a changeset (or two :))!

**Comment by tylerbutler** [comment]
> @znewton thanks so much for adding a changeset (or two :))!

Also the changeset bot is now disabled so it'll stop spamming your PRs.

File: server/routerlicious/packages/services-client/api-report/server-services-client.api.md [hunk 1]
```diff
@@ -647,6 +647,9 @@ export class SummaryTreeUploadManager implements ISummaryUploadManager {
     writeSummaryTree(summaryTree: ISummaryTree_2, parentHandle: string, summaryType: IWholeSummaryPayloadType, sequenceNumber?: number, initial?: boolean): Promise<string>;
 }
 
+// @internal (undocumented)
```
Comment (tylerbutler) [inline server/routerlicious/packages/services-client/api-report/server-services-client.api.md:650]: Can you add a docstring to this constant?

**Review by tylerbutler — APPROVED** [review]
Minor docs nit.

**Pull request merged**